From cdf4ec77f94f4f99a7f4c9a346f3c42d81c80c52 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 16 Jan 2013 03:51:56 +0100
Subject: [PATCH 173/204] shutdown: in the final umount loop don't use
 MNT_FORCE

MNT_FORCE is honoured by NFS and FUSE and allows unmounting of the FS
even if consumers still use it. For our brute-force loop we rely on
EBUSY being reported as long as a file system is still used by a
loopback device or suchlike. Hence, drop MNT_FORCE to make EBUSY
reliable.
(cherry picked from commit 0c08f5cde749bd2818475e487109cd0d413452df)
---
 src/core/umount.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/core/umount.c b/src/core/umount.c
index d97088f..39b61c1 100644
--- a/src/core/umount.c
+++ b/src/core/umount.c
@@ -442,9 +442,11 @@ static int mount_points_list_umount(MountPoint **head, bool *changed, bool log_e
                 )
                         continue;
 
-                /* Trying to umount. Forcing to umount if busy (only for NFS mounts) */
+                /* Trying to umount. We don't force here since we rely
+                 * on busy NFS and FUSE file systems to return EBUSY
+                 * until we closed everything on top of them. */
                 log_info("Unmounting %s.", m->path);
-                if (umount2(m->path, MNT_FORCE) == 0) {
+                if (umount2(m->path, 0) == 0) {
                         if (changed)
                                 *changed = true;
 
-- 
1.8.1.5

